apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,46 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
+import { NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+  Component,
+  EventEmitter,
+  Input,
+  Output,
+  TemplateRef,
+} from '@angular/core';
 import { CardType } from '../../model/card.model';
 import { ListItemComponent } from '../list-item/list-item.component';
 
 @Component({
   selector: 'app-card',
   templateUrl: './card.component.html',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  styles: [
+    `
+      .bg-light-red {
+        background-color: var(--bg-light-red);
+      }
+      .bg-light-green {
+        background-color: var(--bg-light-green);
+      }
+      .bg-light-blue {
+        background-color: var(--bg-light-blue);
+      }
+
+      :host-context(.teacher-theme) .bg-light-red {
+        background-color: var(--bg-light-red);
+      }
+    `,
+  ],
+  imports: [NgIf, NgFor, ListItemComponent, NgTemplateOutlet],
 })
 export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
+  @Input() list: unknown[] | null = null;
   @Input() customClass = '';
+  @Input() specialTemplate!: TemplateRef<unknown>;
```

Comment:
For this kind of reference, @ContentChild is better over @Input. It makes the parent template easier to read